joinservice/internal/kubernetes/kubernetes.go

```diff
@@ -81,22 +82,46 @@ func (c *Client) CreateConfigMap(ctx context.Context, configMap corev1.ConfigMap
 }
 
 // AddNodeToJoiningNodes adds the provided node as a joining node CRD.
-func (c *Client) AddNodeToJoiningNodes(ctx context.Context, nodeName string, componentsHash string) error {
-	joiningNodeResource := schema.GroupVersionResource{Group: "update.edgeless.systems", Version: "v1alpha1", Resource: "joiningnodes"}
-
+func (c *Client) AddNodeToJoiningNodes(ctx context.Context, nodeName string, componentsHash string, isControlPlane bool) error {
 	joiningNode := &unstructured.Unstructured{}
+
+	objectMetadataName := nodeName
+	deadline := metav1.NewTime(time.Now().Add(48 * time.Hour))
```

Comment:
what?! :D

48h seems a bit long doesn't it?

Reply:
Yes. If we were to insert an empty string for `deadline` we could even disable it completely, which was my original idea. But the code path for this in the JoinService is a bit cumbersome. Note that this only applies to worker nodes. Superfluous entries do not create any issues here. So this is basically garbage collection.